server/api/src/rsGeneralAdmin.cpp

```diff
@@ -588,39 +598,110 @@ _addResource(
-            resc_input[irods::RESOURCE_NAME].c_str(),
-            result);
-    }
 
-    // =-=-=-=-=-=-=-
-    // register resource with the metadata catalog, roll back on an error
-    else if ( ( result = chlRegResc( _rsComm, resc_input ) ) != 0 ) {
-        chlRollback( _rsComm );
+        log_api::error("rsGeneralAdmin: acPreProcForCreateResource error for {}, stat={}",
+            resc_input[irods::RESOURCE_NAME], result);
     }
+    // register resource with the metadata catalog, roll back on an error
+    else if ((result = chlRegResc(_rsComm, resc_input)) == 0) {
+        log_api::info("ADDING NEW RESOURCE [{}]", resc_input.at(irods::RESOURCE_NAME));
+        if (const auto err = resc_mgr.init_from_catalog(*_rsComm, resc_input.at(irods::RESOURCE_NAME)); !err.ok()) {
+            log_api::error(err.result());
+        }
 
-    // =-=-=-=-=-=-=-
-    // apply postproc policy enforcement point for creating a resource, handle errors
-    else if ( ( result =  applyRuleArg( "acPostProcForCreateResource", args, argc, &_rei2, NO_SAVE_REI ) ) < 0 ) {
-        if ( _rei2.status < 0 ) {
-            result = _rei2.status;
+        // apply postproc policy enforcement point for creating a resource, handle errors
+        if ( ( result =  applyRuleArg( "acPostProcForCreateResource", args, argc, &_rei2, NO_SAVE_REI ) ) < 0 ) {
+            if ( _rei2.status < 0 ) {
+                result = _rei2.status;
+            }
+
+            log_api::error("rsGeneralAdmin: acPostProcForCreateResource error for {}, stat={}",
+                    resc_input[irods::RESOURCE_NAME], result);
         }
-        log_api::error(
-            "rsGeneralAdmin: acPostProcForCreateResource error for {}, stat={}",
-            resc_input[irods::RESOURCE_NAME].c_str(),
-            result);
+    }
+    else {
+        chlRollback( _rsComm );
     }
 
     return result;
 }
 
+int _remove_resource(RsComm& _comm, generalAdminInp_t& _input, ruleExecInfo_t& _rei2)
+{
+    if (std::strlen(_input.arg2) >= NAME_LEN) {	// resource name
+        return SYS_INVALID_INPUT_PARAM;
+    }
+
+    if (contains_whitespace(_input.arg2)) {
+        log_api::error("Whitespace is not allowed in resource names [{}].", _input.arg2);
+        return CAT_INVALID_RESOURCE_NAME;
+    }
+
+    const std::string_view resc_name = _input.arg2;
+
+    // add a dry run option to idamin rmresc.
+    // basically run chlDelResc then run a rollback immediately after.
+    if (std::strcmp(_input.arg3, "--dryrun") == 0) {
+        log_api::info("Executing a dryrun of removal of resource [{}]", resc_name);
+
+        const auto ec = chlDelResc(&_comm, resc_name.data(), 1);
+
+        if (0 == ec) {
+            log_api::info("DRYRUN REMOVING RESOURCE [{}] :: SUCCESS", _input.arg2);
+        }
+        else {
+            log_api::info("DRYRUN REMOVING RESOURCE [{}] :: FAILURE", _input.arg2);
+        }
+
+        return ec;
+    }
```

Comment:
Is this correct?

We return before making any call to rollback. Perhaps the impl is relying on the ODBC driver to automatically issue a rollback? I hope that isn't a thing.

Reply:
or we're not actually doing any work?

definitely don't want to rely on ODBC driver for a rollback, agreed.